kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/StateStoreOperationRetryManager.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.statestore
+
+import java.util.concurrent.{ConcurrentHashMap, ThreadPoolExecutor, TimeUnit}
+
+import scala.collection.JavaConverters._
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.server.KyuubiServer
+import org.apache.kyuubi.service.AbstractService
+import org.apache.kyuubi.session.KyuubiSessionManager
+import org.apache.kyuubi.util.ThreadUtils
+
+class StateStoreOperationRetryManager private (name: String) extends AbstractService(name)
+  with Logging {
+  def this() = this(classOf[StateStoreOperationRetryManager].getSimpleName)
+
+  private val identifierOperationsRetryRefMap =
+    new ConcurrentHashMap[String, StateOperationsRetryRef]()
+
+  private val retryTrigger =
+    ThreadUtils.newDaemonSingleThreadScheduledExecutor("state-operations-retry-trigger")
+
+  private val retryExecutor: ThreadPoolExecutor =
+    ThreadUtils.newDaemonFixedThreadPool(100, "state-operations-retry-executor")
+
+  private lazy val stateStore = KyuubiServer.kyuubiServer.backendService.sessionManager
```

Comment:
why so hacky?

Reply:
because the sessionManager api will re-add the event.

```
 def updateMetadata(metadata: SessionMetadata): Unit = {
    try {
      sessionStateStore.updateMetadata(metadata)
    } catch {
      case e: Throwable =>
        error(s"Error updating metadata for session ${metadata.identifier}", e)
        val ref =
          stateStoreOperationRetryManager.getOrCreateStateStoreOperationRetryRef(
            metadata.identifier)
        ref.addRetryingSessionStateOperationEvent(RetryingUpdateSessionMetadataEvent(metadata))
    }
  }
```

I can add a parameter `retry:Boolean = true` for that if reuse sessionManager api.